### PR TITLE
Add option for rbfe eq_steps and fix chiral restraint issue

### PR DESCRIPTION
**Pull request by jkausrelay**
- Add option to change the number of eq steps (currently is only 10 ps)
- Fix issue in checking for duplicates of chiral restraints (np array is not hashable)

### CODE REVIEW

File: timemachine/fe/interpolate.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def make_kv(all_idxs, all_params):
 
         def has_duplicates(x):
             x = list(x)
-            return len(set(x)) < len(x)
+            return len(set(to_hashable(x))) < len(x)
```
Comment (mcwitt) [inline timemachine/fe/interpolate.py:90]: Was this change left in unintentionally? (tests still pass without it)

File: timemachine/fe/interpolate.py [hunk 2]
```diff
@@ -138,8 +138,8 @@ def align_chiral_bond_idxs_and_params(src_idxs, src_params, src_signs, dst_idxs,
             dst_idxs,
             zip(dst_signs, dst_params),
             make_default=lambda p: (p[0], 0),  # p[0] is sign, 0 is force constant
-            key=lambda idxs, p: (idxs, p[0]),  # use idxs and sign as key
-            get_idxs=lambda key: key[0],
+            key=lambda idxs, p: (tuple(idxs), p[0]),  # use idxs and sign as key
+            get_idxs=lambda key: tuple(key[0]),
```
Comment (mcwitt) [inline timemachine/fe/interpolate.py:142]: another (maybe nicer?) fix might be to change the definition of `to_hashable` to also convert nested arrays to tuples. This seems to work:
```python
    def to_hashable(x):
        return tuple(to_hashable(e) for e in x) if isinstance(x, Iterable) else x
```

File: timemachine/fe/rbfe.py [hunk 3]
```diff
@@ -431,38 +435,46 @@ def estimate_relative_free_energy(
...
-    try:
-
-        return estimate_free_energy_given_initial_states(
-            initial_states, protocol, temperature, combined_prefix, keep_idxs
-        )
-
-    except Exception as old_exc:
-
-        raise SimulationException(initial_states, protocol, combined_prefix) from old_exc
+    return estimate_free_energy_given_initial_states(initial_states, protocol, temperature, combined_prefix, keep_idxs)
```
Comment (mcwitt) [inline timemachine/fe/rbfe.py:445]: Was this change intentional? Seems like we still want the `SimulationException` to be able to attach additional info?

**Review by mcwitt — CHANGES_REQUESTED** [review]
Thanks for fixing the chiral bond interpolation bug! Looks good overall. Was wondering why we're no longer raising a `SimulationException`?

File: timemachine/fe/interpolate.py (hunk 1 above)
Comment (jkausrelay) [reply timemachine/fe/interpolate.py:90]: reverted. This was my first attempt at a fix, and isn't needed anymore

File: timemachine/fe/interpolate.py (hunk 2 above)
Comment (jkausrelay) [reply timemachine/fe/interpolate.py:142]: updated

File: timemachine/fe/rbfe.py (hunk 3 above)
Comment (jkausrelay) [reply timemachine/fe/rbfe.py]: Updated to store the pickle on exception in rbfe

**Review by mcwitt — APPROVED** [review]
